SparseMatrix/SparseMatrix.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System.Collections;
+
+namespace SparseMatrixApplication
+{
+    public class SparseMatrix : IEnumerable<long>
+    {
+        private readonly int _rowCount;
+        private readonly int _columnCount;
+        private readonly Dictionary<(int, int), long> _elements;
+        private readonly IndexOutOfRangeException _invalidIndexesException = new IndexOutOfRangeException("Invalid indexes.");
+
+        public SparseMatrix(int rowCount, int columnCount)
+        {
+            if (rowCount <= 0 || columnCount <= 0)
+            {
+                throw new ArgumentException("Invalid number of rows or columns.");
+            }
+
+            _rowCount = rowCount;
+            _columnCount = columnCount;
+            _elements = [];
+        }
+
+        public long this[int i, int j]
+        {
+            get
+            {
+                if (!AreIndexesValid(i, j))
+                {
+                    throw _invalidIndexesException;
+                }
+
+                return _elements.TryGetValue((i, j), out long value) ? value : 0;
+            }
+            set
+            {
+                if (!AreIndexesValid(i, j))
+                {
+                    throw _invalidIndexesException;
+                }
+
+                if (value != 0)
+                {
+                    _elements[(i, j)] = value;
+                }
+                else
+                {
+                    _elements.Remove((i, j));
+                }
+            }
+        }
+
+        public override string ToString()
+        {
+            var result = "";
+
+            for (int i = 0; i < _rowCount; i++)
+            {
+                for (int j = 0; j < _columnCount; j++)
+                {
+                    result += this[i, j] + " ";
+                }
+
+                result += "\n";
+            }
+
+            return result;
+        }
+
+        public IEnumerator<long> GetEnumerator()
+        {
+            for (int i = 0; i < _rowCount; i++)
+            {
+                for (int j = 0; j < _columnCount; j++)
+                {
+                    yield return this[i, j];
+                }
+            }
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return GetEnumerator();
+        }
+
+        public IEnumerable<(int, int, long)> GetNonzeroElements()
+        {
```

Comment:
Use linq for the whole result without foreach   (tip-> use Select to form tuple from dictionary)

Reply:
Fixed